examples/interleaved_messages.rs

```diff
@@ -22,8 +22,9 @@ impl Handler<Hello> for ActorA {
 
     async fn handle(&mut self, _: Hello, ctx: &mut Context<Self>) {
         println!("ActorA: Hello");
-        let fut = self.actor_b.send(Hello);
-        ctx.join(self, fut).await.unwrap();
+        xtra::join(ctx.mailbox(), self, self.actor_b.send(Hello))
```

Comment:
Initially, I thought we would get away with not exposing the `Mailbox` through the `Context` but if we don't, these usecases become impossible.